File: root/app/.api/recent-chats.ts
import { NextResponse } from 'next/server';
import { dbService } from '../.db/db-service';

export async function GET(req: Request): Promise<NextResponse> {
    const { searchParams } = new URL(req.url);
    const userId = searchParams.get('userId') || '';

    try {
        const chats = await dbService.messagesConfig.getRecentChats(userId);
        return NextResponse.json(chats);
    } catch(err) {
        return NextResponse.json({ error: 'Failed to load recent chats' }, { status: 500 });
    }
}

File: main/src/main/java/com/app/main/root/app/_api/recent-chats.py
from fastapi import APIRouter, HTTPException
from __main import dbService

router = APIRouter()

@router.get("/api/recent-chats")
async def getRecentChats():
    try:
        chats = await dbService.getRecentChats()
        return chats
    except Exception as err:
        raise HTTPException(status_code=500, detail="Failed to load recent chats!")